src/app/code/community/Shopgate/Cloudapi/Model/Observers/AddSaleRuleTypeToMergedQuote.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/**
+ * Copyright Shopgate Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @author    Shopgate Inc, 804 Congress Ave, Austin, Texas 78701 <interfaces@shopgate.com>
+ * @copyright Shopgate Inc
+ * @license   http://www.apache.org/licenses/LICENSE-2.0 Apache License, Version 2.0
+ */
+
+class Shopgate_Cloudapi_Model_Observers_AddSaleRuleTypeToMergedQuote
+{
+    /**
+     * Transfers the flag for app-only discounts from a guest cart to the customer cart
+     *
+     * @param Varien_Event_Observer $observer
+     */
+    public function execute(Varien_Event_Observer $observer)
+    {
+        if (!Mage::helper('shopgate_cloudapi/request')->isShopgateApi()) {
+            return;
+        }
+
+        $oldQuoteId = $observer->getData('source')->getId();
+        $newQuoteId = $observer->getData('quote')->getId();
+        /** @var Shopgate_Cloudapi_Model_Resource_Cart_Source_Collection $collection */
+        $collection = Mage::getResourceModel('shopgate_cloudapi/cart_source_collection')
+                          ->addFieldToFilter(array('quote_id'), array(array($newQuoteId, $oldQuoteId)))
+                          ->addFieldToFilter('source', Shopgate_Cloudapi_Model_Cart_Source::SOURCE_SHOPGATE_APP);
+        if (null !== $this->getQuoteById($collection, $oldQuoteId)
```

Comment:
@Skistims I just have a question regarding this line of code. Why was it necessary to check if the old quote id exists in our `cart_sources` DB? Old quote meaning guest quote.

Reply:
never mind, I realized my mistake. The merge actually happens in the desktop of magento once the user registers, not our endpoint. This requires the logic to be in the observer. I also see the guest check is a safe-guard to see if it is our transaction. I will not omit the guest check as it's now more performant. I also added the check for our inApp cart cookie to be more safe.